src/toolchain/toolchain.ts

```diff
@@ -85,42 +84,28 @@ export class SwiftToolchain {
         return undefined;
     }
 
-    /**
-     * @returns path to developer folder, where we find XCTest
-     */
-    private static async getDeveloperDir(): Promise<string | undefined> {
-        try {
-            switch (process.platform) {
-                case "darwin": {
-                    const { stdout } = await execFile("xcode-select", ["-p"]);
-                    return stdout.trimEnd();
-                }
-
-                case "win32": {
-                    const developerDir = process.env.DEVELOPER_DIR;
-                    if (!developerDir) {
-                        throw Error("Environment variable DEVELOPER_DIR is not set.");
-                    }
-                    return developerDir;
-                }
-            }
-            return undefined;
-        } catch {
-            return undefined;
+    private static getSDKROOT(): string | undefined {
+        if (process.platform === "win32") {
+            return process.env.SDKROOT ?? undefined;
         }
+        return undefined;
     }
 
     /**
      * @param developerDir Developer directory
      * @returns Path to folder where xctest can be found
      */
-    private static async getXCTestPath(developerDir: string): Promise<string | undefined> {
+    private static async getXCTestPath(sdkroot: string | undefined): Promise<string | undefined> {
         switch (process.platform) {
-            case "darwin":
-                return path.join(developerDir, "usr", "bin");
-
+            case "darwin": {
+                const { stdout } = await execFile("xcode-select", ["-p"]);
+                return path.join(stdout.trimEnd(), "usr", "bin");
+            }
             case "win32": {
-                const platformPath = path.join(developerDir, "Platforms", "Windows.platform");
+                if (!sdkroot) {
+                    return undefined;
```

Comment:
Should we allow the extension to startup if someone hasn't got an SDKROOT environment variable. Tests won't work. Should we throw an Error here instead. VSCode will catch it and display the error. Something like
```ts
throw Error("SDKROOT environment variable should be set");
```

Reply:
My personal opinion: we should.  The `SDKROOT` environment variable is just a convenience thing.  If you do not have that set (and I often do not), I expect that the user has intentionally not installed or uninstalled the build platform's SDK.  In such a case, they would need _some_ SDK and the assumption is that they are compiling to a foreign platform.  That is to say, this is setup in such a way that if I do not setup `SDKROOT`, I can simply pass `-sdk C:\Library\Developer\Platforms\Android.platform\Developer\SDKs\Android.sdk -target armv7-unknown-linux-androideabi` and use the toolchain to build for android :)

Reply:
ok